service/storage/operations.go

```diff
@@ -173,18 +170,17 @@ func (l *Library) RetrieveEvents(height uint64, types []flow.EventType, events *
 	}
 }
 
-// RetrievePayload retrieves the ledger payloads at the given height that match the given path.
-func (l *Library) RetrievePayload(height uint64, path ledger.Path, payload *ledger.Payload) func(*badger.Txn) error {
+// RetrievePayload retrieves the ledger payloads at the given height that match the given registerID.
+func (l *Library) RetrievePayload(height uint64, register flow.RegisterID, payload *flow.RegisterValue) func(*badger.Txn) error {
 	return func(tx *badger.Txn) error {
-
-		key := EncodeKey(PrefixPayload, path, height)
+		key := EncodeKey(PrefixPayload, register, height)
 		it := tx.NewIterator(badger.IteratorOptions{
 			PrefetchSize:   0,
 			PrefetchValues: false,
 			Reverse:        true,
 			AllVersions:    false,
 			InternalAccess: false,
-			Prefix:         key[:1+pathfinder.PathByteSize],
+			Prefix:         key[:len(key)-heightBinarySize],
```

Comment:
```suggestion
			// In order to find the latest payload up to the given height, we need to iterate
			// through all the keys for the registers on different height.
			// so the prefix is the encoded register ID
			Prefix:         key[:len(key)-heightBinarySize],
```

Reply:
Here we first encode the register with the height, and later we subtract the bytes for the height. 

Would it be more clear if we just encode the register ID?

```
  Prefix: EncodeRegisterID(register)
```